### PR TITLE
[DOCS] added several small docs

### PR DESCRIPTION
**Pull request by Grosskopf**
Mostly basic mathematic stuff and the curve stuff we used in our small Jamsession yesterday :)

### CODE REVIEW

File: doc/classes/Path2D.xml [hunk 1]
```diff
@@ -30,6 +30,7 @@
 	</methods>
 	<members>
 		<member name="curve" type="Curve2D" setter="set_curve" getter="get_curve">
+			The path information as a [Curve2D].
```
Comment (cbscribe) [inline doc/classes/Path2D.xml]: A [Curve2D] describing the path.

File: doc/classes/Path2D.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="Path2D" inherits="Node2D" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
-		Container for a [Curve2D].
+		Path to move nodes across on [PathFollow2D] nodes. Container for a [Curve2D].
 	</brief_description>
 	<description>
 		This class is a container/Node-ification of a [Curve2D], so it can have [Node2D] properties and [Node] info.
```
Comment (cbscribe) [inline doc/classes/Path2D.xml]: Yikes!

File: doc/classes/Quat.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
 		Quaternion.
 	</brief_description>
 	<description>
-		Quaternion is a 4 dimensional vector that is used to represent a rotation. It mainly exists to perform SLERP (spherical-linear interpolation) between two rotations. Multiplying quaternions also cheaply reproduces rotation sequences. However quaternions need to be often renormalized, or else they suffer from precision issues.
+		A 4 dimensional vector that is used to represent a rotation. It mainly exists to perform SLERP (spherical-linear interpolation) between two rotations. Multiplying quaternions also cheaply reproduces rotation sequences. However quaternions need to be often renormalized, or else they suffer from precision issues.
```
Comment (cbscribe) [inline doc/classes/Quat.xml]: A 4-dimensional vector representing a rotation.

This needs further rewrite, removing "mainly exists" and "cheaply". Quaternions are a very difficult concept, and we need a nice, clear explanation here.

File: doc/classes/Quat.xml [hunk 4]
```diff
@@ -122,6 +124,7 @@
 			<argument index="1" name="t" type="float">
 			</argument>
 			<description>
+				Performs a spherical-linear interpolation with another quaterion without checking if the rotationpath is the short one.
```
Comment (cbscribe) [inline doc/classes/Quat.xml]: - rotation path
- shortest

File: doc/classes/Quat.xml [hunk 5]
```diff
@@ -130,17 +133,22 @@
 			<argument index="0" name="v" type="Vector3">
 			</argument>
 			<description>
+				Transforms the vector [code]v[/code] with this quaternion.
```
Comment (cbscribe) [inline doc/classes/Quat.xml]: - with -> by

File: doc/classes/Rect2.xml [hunk 6]
```diff
@@ -34,7 +34,7 @@
 			<argument index="3" name="height" type="float">
 			</argument>
 			<description>
-				Construct a [code]Rect2[/code] by x, y, width and height.
+				Constructs a [code]Rect2[/code] by x, y, width and height.
```
Comment (cbscribe) [inline doc/classes/Rect2.xml]: - Missing Oxford comma

File: doc/classes/Path2D.xml [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="Path2D" inherits="Node2D" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
-		Container for a [Curve2D].
+		Path to move nodes across on [PathFollow2D] nodes. Container for a [Curve2D].
```
Comment (cbscribe) [inline doc/classes/Path2D.xml]: Contains a [Curve2D] path for [PathFollow2D] nodes to follow.

File: doc/classes/Rect2.xml [hunk 8]
```diff
@@ -61,14 +61,14 @@
...
-				Return this [code]Rect2[/code] expanded to include a given point.
+				Returns this [code]Rect2[/code] expanded to include a given point.
 			</description>
 		</method>
 		<method name="get_area">
 			<return type="float">
 			</return>
 			<description>
-				Get the area of the [code]Rect2[/code].
+				Gets the area of the [code]Rect2[/code].
```
Comment (cbscribe) [inline doc/classes/Rect2.xml]: Gets -> Returns

**Review by cbscribe — CHANGES_REQUESTED** [review]
Didn't get to Rect2 & Rect3 yet, but here's a bunch of changes/comments

File: doc/classes/Node2D.xml [hunk 9]
```diff
@@ -163,7 +163,7 @@
 			<argument index="0" name="radians" type="float">
 			</argument>
 			<description>
-				Apply a rotation to the node, in radians, starting from its current rotation.
+				Applies a rotation to the node, in radians, starting from its current rotation.
```
[the remark this answers is not in the document]
Comment (Grosskopf) [reply doc/classes/Node2D.xml]: is this a problem? I went past it and added the "s" without thinking much, the set/get functions will be removed anyway right?

File: doc/classes/Quat.xml (hunk 3 above)
Comment (Grosskopf) [reply doc/classes/Quat.xml]: i totally agree

File: doc/classes/Node2D.xml (hunk 9 above)
[the remark this answers is not in the document]
Comment (cbscribe) [reply doc/classes/Node2D.xml]: Yeah, it won't really matter. I was more hinting that if you don't change them, we don't have to review them. :P

File: doc/classes/Vector3.xml [hunk 10]
```diff
@@ -61,7 +62,7 @@
 			<argument index="0" name="b" type="Vector3">
 			</argument>
 			<description>
-				Return the cross product with b.
+				Returns the cross product with b.
```
Comment (cbscribe) [inline doc/classes/Vector3.xml]: [code]b[/code]

File: doc/classes/Vector3.xml [hunk 11]
```diff
@@ -76,7 +77,7 @@
 			<argument index="3" name="t" type="float">
 			</argument>
 			<description>
-				Perform a cubic interpolation between vectors pre_a, a, b, post_b (a is current), by the given amount (t).
+				Performs a cubic interpolation between vectors pre_a, a, b, post_b (a is current), by the given amount (t).
```
Comment (cbscribe) [inline doc/classes/Vector3.xml]: put [code] tags around the arguments

File: doc/classes/Vector3.xml [hunk 12]
```diff
@@ -85,7 +86,7 @@
 			<argument index="0" name="b" type="Vector3">
 			</argument>
 			<description>
-				Return the squared distance (distance minus the last square root) to b. Prefer this function over distance_to if you need to sort vectors or need the squared distance for some formula.
+				Returns the squared distance (distance minus the last square root) to b. Prefer this function over distance_to if you need to sort vectors or need the squared distance for some formula.
```
Comment (cbscribe) [inline doc/classes/Vector3.xml]: - "minus the last square root"? That doesn't make sense. "Returns the squared distance to [code]b[/code]."

File: doc/classes/Rect3.xml [hunk 13]
```diff
@@ -183,7 +183,7 @@
 			<argument index="0" name="with" type="Rect3">
 			</argument>
 			<description>
-				Combine this [code]Rect3[/code] with another, a larger one is returned that contains both.
+				Combines this [code]Rect3[/code] with another, a larger one is returned that contains both.
```
Comment (cbscribe) [inline doc/classes/Rect3.xml]: Why not just:

Returns a larger Rect3 that contains this Rect3 and [code]with[/code].

File: doc/classes/Rect3.xml [hunk 14]
```diff
@@ -174,7 +174,7 @@
 			<argument index="1" name="to" type="Vector3">
 			</argument>
 			<description>
-				Return true if the [code]Rect3[/code] intersects the line segment between from and to
+				Returns true if the [code]Rect3[/code] intersects the line segment between from and to.
```
Comment (cbscribe) [inline doc/classes/Rect3.xml]: - code tags: true, from, to

File: doc/classes/Rect3.xml [hunk 15]
```diff
@@ -163,7 +163,7 @@
 			<argument index="0" name="plane" type="Plane">
 			</argument>
 			<description>
-				Return true if the [code]Rect3[/code] is at both sides of a plane.
+				Returns true if the [code]Rect3[/code] is at both sides of a plane.
```
Comment (cbscribe) [inline doc/classes/Rect3.xml]: - code tags around true
- at -> on

File: doc/classes/Rect3.xml [hunk 16]
```diff
@@ -154,7 +154,7 @@
 			<argument index="0" name="with" type="Rect3">
 			</argument>
 			<description>
-				Return true if the [code]Rect3[/code] overlaps with another.
+				Returns true if the [code]Rect3[/code] overlaps with another.
```
Comment (cbscribe) [inline doc/classes/Rect3.xml]: - code tags around true

File: doc/classes/Rect3.xml [hunk 17]
```diff
@@ -136,7 +136,7 @@
 			<argument index="0" name="point" type="Vector3">
 			</argument>
 			<description>
-				Return true if the [code]Rect3[/code] contains a point.
+				Returns true if the [code]Rect3[/code] contains a point.
```
Comment (cbscribe) [inline doc/classes/Rect3.xml]: - code tags around true

File: doc/classes/Rect3.xml [hunk 18]
```diff
@@ -113,21 +113,21 @@
...
-				Return true if the [code]Rect3[/code] is flat or empty.
+				Returns true if the [code]Rect3[/code] is flat or empty.
 			</description>
 		</method>
 		<method name="has_no_surface">
 			<return type="bool">
 			</return>
 			<description>
-				Return true if the [code]Rect3[/code] is empty.
+				Returns true if the [code]Rect3[/code] is empty.
```
Comment (cbscribe) [inline doc/classes/Rect3.xml]: - code tags around true

File: doc/classes/Rect3.xml [hunk 19]
```diff
@@ -113,21 +113,21 @@
...
-				Return a copy of the [code]Rect3[/code] grown a given amount of units towards all the sides.
+				Returns a copy of the [code]Rect3[/code] grown a given amount of units towards all the sides.
 			</description>
 		</method>
 		<method name="has_no_area">
 			<return type="bool">
 			</return>
 			<description>
-				Return true if the [code]Rect3[/code] is flat or empty.
+				Returns true if the [code]Rect3[/code] is flat or empty.
```
Comment (cbscribe) [inline doc/classes/Rect3.xml]: - code tags around true

File: doc/classes/Rect3.xml [hunk 20]
```diff
@@ -28,7 +28,7 @@
 			<argument index="0" name="with" type="Rect3">
 			</argument>
 			<description>
-				Return true if this [code]Rect3[/code] completely encloses another one.
+				Returns true if this [code]Rect3[/code] completely encloses another one.
```
Comment (cbscribe) [inline doc/classes/Rect3.xml]: - code tags around true

File: doc/classes/Rect3.xml [hunk 21]
```diff
@@ -4,7 +4,7 @@
 		Axis-Aligned Bounding Box.
 	</brief_description>
 	<description>
-		Rect3 provides an 3D Axis-Aligned Bounding Box. It consists of a position, a size, and several utility functions. It is typically used for simple (fast) overlap tests.
+		Rect3 consists of a position, a size, and several utility functions. It is typically used for simple (fast) overlap tests.
```
Comment (cbscribe) [inline doc/classes/Rect3.xml]: Don't use "simple" (fast is OK)

File: doc/classes/Rect2.xml [hunk 22]
```diff
@@ -4,7 +4,7 @@
 		2D Axis-aligned bounding box.
 	</brief_description>
 	<description>
-		Rect2 provides an 2D Axis-Aligned Bounding Box. It consists of a position, a size, and several utility functions. It is typically used for fast overlap tests.
+		Rect2 consists of a position, a size, and several utility functions. It is typically used for fast overlap tests.
```
Comment (cbscribe) [inline doc/classes/Rect2.xml]: See Rect3 comments. Almost all the same comments apply (code tags, etc).

**Review by cbscribe — CHANGES_REQUESTED** [review]

**Review by cbscribe — APPROVED** [review]

**Comment by cbscribe** [comment]
Wow, that was a big one!

**Pull request merged**